Crash when using data_disk in azurerm_image

<!--- Please keep this note for the community --->

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

<!--- Thank you for keeping this note for the community --->

### Terraform Version
Terraform v0.11.7
+ provider.azurerm v1.11.0

<!--- Please run `terraform -v` to show the Terraform core version and provider version(s). If you are not running the latest version of Terraform or the provider, please upgrade because your issue may have already been fixed. [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions). --->

### Affected Resource(s)

<!--- Please list the affected resources and data sources. --->
* azurerm_image

### Terraform Configuration Files

<!--- Information about code formatting: https://help.github.com/articles/basic-writing-and-formatting-syntax/#quoting-code --->

```hcl
# Copy-paste your Terraform configurations here - for large Terraform configs,
# please use a service like Dropbox and share a link to the ZIP file. For
# security, you can also encrypt the files using our GPG public key: https://keybase.io/hashicorp
provider "azurerm" {}

resource "azurerm_resource_group" "test" {
  name     = "Terraform-issue"
  location = "West US"
}

resource "azurerm_image" "test" {
  name = "acctest"
  location = "West US"
  resource_group_name = "${azurerm_resource_group.test.name}"

  os_disk {
    os_type = "Linux"
    os_state = "Generalized"
    blob_uri = "<redacted_blob_uri>"
  }
  data_disk {
   lun = "0"
   blob_uri = "<redacted_blob_uri>"   
  }
}

```

### Debug Output

<!---
Please provide a link to a GitHub Gist containing the complete debug output. Please do NOT paste the debug output in the issue; just paste a link to the Gist.

To obtain the debug output, see the [Terraform documentation on debugging](https://www.terraform.io/docs/internals/debugging.html).
--->

### Panic Output

<!--- If Terraform produced a panic, please provide a link to a GitHub Gist containing the output of the `crash.log`. --->
panic: interface conversion: interface {} is nil, not string
2018-07-26T08:56:04.885-0600 [DEBUG] plugin.terraform-provider-azurerm_v1.11.0_x4: 
2018-07-26T08:56:04.886-0600 [DEBUG] plugin.terraform-provider-azurerm_v1.11.0_x4: goroutine 93 [running]:
2018-07-26T08:56:04.886-0600 [DEBUG] plugin.terraform-provider-azurerm_v1.11.0_x4: github.com/terraform-providers/terraform-provider-azurerm/azurerm.expandAzureRmImageDataDisks(0xc420696460, 0xc42063aa60, 0x0, 0x0, 0xc42071c240, 0x0)
2018-07-26T08:56:04.886-0600 [DEBUG] plugin.terraform-provider-azurerm_v1.11.0_x4:      /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/azurerm/resource_arm_image.go:411 +0x64e
2018-07-26T08:56:04.886-0600 [DEBUG] plugin.terraform-provider-azurerm_v1.11.0_x4: github.com/terraform-providers/terraform-provider-azurerm/azurerm.resourceArmImageCreateUpdate(0xc420696460, 0x1e2aea0, 0xc42044e000, 0xc420696460, 0x0)
2018-07-26T08:56:04.886-0600 [DEBUG] plugin.terraform-provider-azurerm_v1.11.0_x4:      /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/azurerm/resource_arm_image.go:178 +0x35b
2018-07-26T08:56:04.886-0600 [DEBUG] plugin.terraform-provider-azurerm_v1.11.0_x4: github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform/helper/schema.(*Resource).Apply(0xc4207ed2d0, 0xc420668af0, 0xc42063a820, 0x1e2aea0, 0xc42044e000, 0x1, 0xc4204e1980, 0xc4204e1980)
2018-07-26T08:56:04.886-0600 [DEBUG] plugin.terraform-provider-azurerm_v1.11.0_x4:      /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform/helper/schema/resource.go:227 +0x364
2018-07-26T08:56:04.886-0600 [DEBUG] plugin.terraform-provider-azurerm_v1.11.0_x4: github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform/helper/schema.(*Provider).Apply(0xc420239b20, 0xc420668aa0, 0xc420668af0, 0xc42063a820, 0x7fdcb70e3000, 0x0, 0x0)
2018-07-26T08:56:04.886-0600 [DEBUG] plugin.terraform-provider-azurerm_v1.11.0_x4:      /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform/helper/schema/provider.go:283 +0xa4
2018-07-26T08:56:04.886-0600 [DEBUG] plugin.terraform-provider-azurerm_v1.11.0_x4: github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform/plugin.(*ResourceProviderServer).Apply(0xc4207c46a0, 0xc42063a7c0, 0xc4204da410, 0x0, 0x0)
2018-07-26T08:56:04.886-0600 [DEBUG] plugin.terraform-provider-azurerm_v1.11.0_x4:      /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform/plugin/resource_provider.go:527 +0x57
2018-07-26T08:56:04.887-0600 [DEBUG] plugin.terraform-provider-azurerm_v1.11.0_x4: reflect.Value.call(0xc4203e82a0, 0xc42000c008, 0x13, 0x1e681f4, 0x4, 0xc4205b5f20, 0x3, 0x3, 0x0, 0x0, ...)
2018-07-26T08:56:04.887-0600 [DEBUG] plugin.terraform-provider-azurerm_v1.11.0_x4:      /usr/local/go/src/reflect/value.go:434 +0x905
2018-07-26T08:56:04.887-0600 [DEBUG] plugin.terraform-provider-azurerm_v1.11.0_x4: reflect.Value.Call(0xc4203e82a0, 0xc42000c008, 0x13, 0xc4201f9f20, 0x3, 0x3, 0xc420428f00, 0x7ae8b7, 0xc420428c00)
2018-07-26T08:56:04.887-0600 [DEBUG] plugin.terraform-provider-azurerm_v1.11.0_x4:      /usr/local/go/src/reflect/value.go:302 +0xa4
2018/07/26 08:56:04 [TRACE] root: eval: *terraform.EvalWriteState
2018/07/26 08:56:04 [TRACE] root: eval: *terraform.EvalApplyProvisioners
2018/07/26 08:56:04 [TRACE] root: eval: *terraform.EvalIf
2018/07/26 08:56:04 [TRACE] root: eval: *terraform.EvalWriteState
2018/07/26 08:56:04 [TRACE] root: eval: *terraform.EvalWriteDiff
2018/07/26 08:56:04 [TRACE] root: eval: *terraform.EvalApplyPost
2018/07/26 08:56:04 [ERROR] root: eval: *terraform.EvalApplyPost, err: 1 error(s) occurred:

* azurerm_image.test: unexpected EOF
2018/07/26 08:56:04 [ERROR] root: eval: *terraform.EvalSequence, err: 1 error(s) occurred:

* azurerm_image.test: unexpected EOF
2018/07/26 08:56:04 [TRACE] [walkApply] Exiting eval tree: azurerm_image.test
2018/07/26 08:56:04 [TRACE] dag/walk: upstream errored, not walking "provider.azurerm (close)"
2018/07/26 08:56:04 [TRACE] dag/walk: upstream errored, not walking "meta.count-boundary (count boundary fixup)"
2018/07/26 08:56:04 [TRACE] dag/walk: upstream errored, not walking "root"
2018-07-26T08:56:04.889-0600 [DEBUG] plugin.terraform-provider-azurerm_v1.11.0_x4: net/rpc.(*service).call(0xc420638200, 0xc420668000, 0xc420c16050, 0xc4208d2280, 0xc4207c47a0, 0x1b336a0, 0xc42063a7c0, 0x16, 0x1b336e0, 0xc4204da410, ...)
2018-07-26T08:56:04.889-0600 [DEBUG] plugin.terraform-provider-azurerm_v1.11.0_x4:      /usr/local/go/src/net/rpc/server.go:381 +0x142
2018-07-26T08:56:04.889-0600 [DEBUG] plugin.terraform-provider-azurerm_v1.11.0_x4: created by net/rpc.(*Server).ServeCodec
2018-07-26T08:56:04.889-0600 [DEBUG] plugin.terraform-provider-azurerm_v1.11.0_x4:      /usr/local/go/src/net/rpc/server.go:475 +0x36b
2018-07-26T08:56:04.889-0600 [DEBUG] plugin: plugin process exited: path=/media/sf_code/terraform/bug/.terraform/plugins/linux_amd64/terraform-provider-azurerm_v1.11.0_x4
2018/07/26 08:56:04 [TRACE] Preserving existing state lineage "f9e73667-861a-cd67-f7fd-a0211d470e32"
2018/07/26 08:56:04 [TRACE] Preserving existing state lineage "f9e73667-861a-cd67-f7fd-a0211d470e32"
2018/07/26 08:56:04 [TRACE] Preserving existing state lineage "f9e73667-861a-cd67-f7fd-a0211d470e32"
2018/07/26 08:56:04 [DEBUG] plugin: waiting for all plugin processes to complete...
2018-07-26T08:56:04.902-0600 [WARN ] plugin: error closing client during Kill: err="connection is shut down"



!!!!!!!!!!!!!!!!!!!!!!!!!!! TERRAFORM CRASH !!!!!!!!!!!!!!!!!!!!!!!!!!!!

Terraform crashed! This is always indicative of a bug within Terraform.
A crash log has been placed at "crash.log" relative to your current
working directory. It would be immensely helpful if you could please
report the crash with Terraform[1] so that we can fix this.

When reporting bugs, please include your terraform version. That
information is available on the first line of crash.log. You can also
get it by running 'terraform --version' on the command line.

[1]: https://github.com/hashicorp/terraform/issues

!!!!!!!!!!!!!!!!!!!!!!!!!!! TERRAFORM CRASH !!!!!!!!!!!!!!!!!!!!!!!!!!!!

### Expected Behavior

Image should have been created with a data_disk

### Actual Behavior

terraform crash

### Steps to Reproduce

<!--- Please list the steps required to reproduce the issue. --->

1. Create an image block with a data_disk
2. `terraform apply`

### Important Factoids

<!--- Are there anything atypical about your accounts that we should know? For example: Running in a Azure China/Germany/Government? --->

### References

<!---
Information about referencing Github Issues: https://help.github.com/articles/basic-writing-and-formatting-syntax/#referencing-issues-and-pull-requests

Are there any other GitHub issues (open or closed) or pull requests that should be linked here? Such as vendor documentation?
--->

* #0000

--- a/azurerm/resource_arm_image.go
+++ b/azurerm/resource_arm_image.go
@@ -395,7 +395,6 @@ func expandAzureRmImageOsDisk(d *schema.ResourceData) (*compute.ImageOSDisk, err
 			osState := compute.OperatingSystemStateTypes(v)
 			osDisk.OsState = osState
 		}
-
 		managedDiskID := config["managed_disk_id"].(string)
 		if managedDiskID != "" {
 			managedDisk := &compute.SubResource{

--- a/azurerm/resource_arm_image.go
+++ b/azurerm/resource_arm_image.go
@@ -429,8 +429,9 @@ func expandAzureRmImageDataDisks(d *schema.ResourceData) ([]compute.ImageDataDis
 	for _, diskConfig := range disks {
 		config := diskConfig.(map[string]interface{})
 
-		managedDiskID := d.Get("managed_disk_id").(string)
-		blobURI := d.Get("blob_uri").(string)
+		managedDiskID := config["managed_disk_id"].(string)
+
+		blobURI := config["blob_uri"].(string)
 		lun := int32(config["lun"].(int))
 
 		dataDisk := compute.ImageDataDisk{
@@ -438,12 +439,12 @@ func expandAzureRmImageDataDisks(d *schema.ResourceData) ([]compute.ImageDataDis
 			BlobURI: &blobURI,
 		}
 
-		if size := d.Get("size_gb"); size != 0 {
+		if size := config["size_gb"]; size != 0 {
 			diskSize := int32(size.(int))
 			dataDisk.DiskSizeGB = &diskSize
 		}
 
-		if v := d.Get("caching").(string); v != "" {
+		if v := config["caching"].(string); v != "" {
 			caching := compute.CachingTypes(v)
 			dataDisk.Caching = caching
 		}

